core/ra2.js

```diff
@@ -365,6 +388,13 @@ export default class RA2State {
         serverPublickey.set(serverKeyLengthBuffer);
         serverPublickey.set(serverN, 4);
         serverPublickey.set(serverE, 4 + serverKeyBytes);
+
+        // verify server public key
+        this.dispatchEvent(new CustomEvent("serverkeyverification", {
+            detail: { serverPublickey: serverPublickey }
```

Comment:
Sound reasonable. But then we need to include the type somewhere so that the UI knows how to handle them when other types will be added. E.g. elliptic curves.